File: python/matrix/matrix.py
class Matrix:
    M = []

    def __init__(self, matrix_string):
        M = []
        rows = matrix_string.split("\n")
        for row in rows:
            cells = row.split(" ")
            m = []
            for cell in cells:
                m.append(int(cell))
            M.append(m)

    def row(self, index):
        return M[index]

    def column(self, index):
        return M[index]

File: java/matrix/src/main/java/Matrix.java
import java.util.Arrays;

class Matrix {

    private final int[][] m;

    Matrix(String matrixAsString) {
        this.m = Arrays.stream(matrixAsString.split("\n"))
                .map(row -> Arrays.stream(row.split(" "))
                        .mapToInt(Integer::parseInt)
                        .toArray())
                .toArray(int[][]::new);
    }

    int[] getRow(int rowNumber) {
        return m[rowNumber - 1];
    }

    int[] getColumn(int columnNumber) {
        return Arrays.stream(m)
                .mapToInt(row -> row[columnNumber - 1])
                .toArray();
    }
}
